cxx/irm.cc

```diff
@@ -21,7 +22,9 @@ RelationVariant clean_relation_from_spec(const std::string& name,
     case ObservationEnum::string_type:
       return new CleanRelation<std::string>(name, spec, doms);
     default:
-      assert(false && "Unsupported observation type.");
+      printf("Unknown observation type %d for relation %s\n",
+             (int)(spec.observation_type), name.c_str());
+      std::exit(1);
```

Comment:
I don't have a good understanding of the differences between `assert(false)`, `exit(1)`, and `abort()` in terms of whether destructors are called, whether there's a core dump, etc. Is there anything we need to consider there?

Reply:
Thanks, I see (I was thinking of this https://learn.microsoft.com/en-us/cpp/cpp/program-termination?view=msvc-170, which talks about differences in whether global object destructors are called.)